webapp/channels/src/components/status_dropdown/status_dropdown.tsx

```diff
@@ -342,7 +342,7 @@ export class StatusDropdown extends React.PureComponent<Props, State> {
                     time={customStatus.expires_at}
                     timezone={this.props.timezone}
                     className={classNames('custom_status__expiry', {
-                        padded: customStatus?.text.length > 0,
+                        padded: customStatus?.text?.length > 0,
```

Comment:
Are you sure we need to handle `customStatus.text` or `customStatus.emoji` being undefined? Based on `model/custom_status.go`, they should at least be empty strings

Reply:
After this PR, this code would no longer be necessary. This was done to handle existing bad data.

Reply:
Should there be a migration to fix any existing cases of this then? Or could we just assume that there are no existing cases of this since anyone with it would need to manually fix it if there's no automatic migration.

We have a lot of random null checking on the client and a ton of `?.`s spread all over the frontend code, so I'd like to avoid having to null check the custom status since we have this validation now

Reply:
In that case, I think it'd make sense to revert the changes to this file. While we could guard against this on the frontend, I'd rather assume that the data that we get from the server is always valid and then type it accordingly. As it is, we validate the data on the server and then we validate it again on the frontend which feels unnecessary.

I'm only 1/5 on that though since user props aren't currently typed in a way that any of it is type safe, so feel free to disregard this